tests/gateway/helper_functions.py

```diff
@@ -0,0 +1,88 @@
+from pathlib import Path
+import subprocess
+import sys
+import time
+from typing import Any, Union
+
+import requests
+import yaml
+
+import mlflow.gateway.utils
+from mlflow.gateway.utils import kill_child_processes
+from tests.helper_functions import get_safe_port
+
+
+def reset_gateway_uri():
```

Comment:
@harupy how do you feel about this? Is there a 'cleaner' way that you'd prefer us to reset this global variable between tests?

Reply:
We should use an autouse fixture.